library/Sexp/src/Juvix/Sexp/Structure/CoreNamed.hs

```diff
@@ -0,0 +1,441 @@
+-- |
+-- Hardeing S-expressions into a more readable form. Here we use a
+-- mixture of record structures and aliases. Each cover a form that we
+-- wish to talk about rather than just match away at
+--
+-- - _The form for transformation follows this structure_
+-- #+begin_src haskell
+--   -- the data type
+--   data Form = ... deriving (Show)
+--   is<Form>   :: Sexp.T -> Bool
+--   to<Form>   :: Sexp.T -> Maybe <Form>
+--   from<Form> :: <Form> -> Sexp.T
+-- #+end_src
+-- + With the following properties of the forms
+--   #+begin_src haskell
+--     ∀ s : Sexp.T. is<Form> s = True ⟷ is-just (to<Form> s)
+--
+--     to<Form> 。 from<Form> = Just
+--   #+end_src
+-- _TODO_
+--  1. Figure out if we can even express a spec system in
+--     Haskell... =to<Form>= and =From<From>= have the exact same signature
+--  2. replace the repeat code with the =to<Form>= with an abstraction
+--  3. put the meta data with the form so we don't have to do it by
+--     hand in the code that uses this
+--     1. Use =Juvix.Library.LineNum=
+--     2. append the =Form= With this
+--     3. have to<Form> fill this
+--     4. Have extra smart consturctors that are =<form>=, so that we
+--        can automatically fill in this meta data
+module Juvix.Sexp.Structure.CoreNamed where
+
+import Juvix.Library hiding (Meta, fromInteger, toInteger)
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified Juvix.Sexp as Sexp
+import Juvix.Sexp.Structure.Helpers
+
+newtype Star = Star {starUniverse :: Integer} deriving (Show)
```

Comment:
Why don't we use the types from Core.Base directly, instead of creating new ones here?

Reply:
Sadly targeting `Core.Base` is hard in the sense that we have many unsupported forms, and this leads into targetting them easier!